juju/controller.py

```diff
@@ -529,40 +529,50 @@ async def get_models(self, all_=False, username=None):
         .. deprecated:: 0.7.0
            Use :meth:`.list_models` instead.
         """
-        controller_facade = client.ControllerFacade.from_connection(
-            self.connection())
-        for attempt in (1, 2, 3):
-            try:
-                return await controller_facade.AllModels()
-            except errors.JujuAPIError as e:
-                # retry concurrency error until resolved in Juju
-                # see: https://bugs.launchpad.net/juju/+bug/1721786
-                if 'has been removed' not in e.message or attempt == 3:
-                    raise
+        return await self.list_models(username, all_)
+
+    async def model_uuids(self, username=None, all_=False):
+        """Return a mapping of model names to UUIDs the given user can access.
+
+        :param str username: Optional username argument, defaults to
+        current connected user.
 
-    async def model_uuids(self):
-        """Return a mapping of model names to UUIDs.
+        :param bool all_: Flag to list all models, regardless of
+        user accessibility (administrative users only)
+
+        :returns: {str name : str UUID}
         """
-        controller_facade = client.ControllerFacade.from_connection(
+
+        facade = client.ModelManagerFacade.from_connection(
             self.connection())
+        if all_:
+            facade = client.ControllerFacade.from_connection(
+                self.connection())
```

Comment:
I think this should be an `if all_: ... else ...` So we don't create a facade to only throw it away.